polynote-frontend/polynote/util/helpers.ts

```diff
@@ -353,6 +353,26 @@ export function positionIn(str: string, line: number, column: number): number {
     return pos;
 }
 
+/**
+ * Helper method for grabbing everything in a string up to the nth occurrence of the pattern.
+ * Will only return a nonempty string for n >= 1.
+ * E.g. if called with "hello/world" where n = 1 and pattern = "/", will return "hello"
+ */
+export function getUpToNthOccurrence(str: string, n: number, pattern: string): string {
```

Comment:
Is this different from the optional `limit` parameter on `split`? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split#syntax

Reply:
I _think_ it's different? What I was specifically looking for (and which I didn't make very clear from the comment) is to _also_ return the pattern in the string. Traversing the file tree, I needed to get everything up to a certain slash, so for example, wanting to get the first few folders in a path like `root/folder/inner-folder/file.ipynb`, we might want `root/folder` or `root/folder/inner-folder`.

Reply:
Maybe I'm missing something. I think this does what you want?
![image](https://github.com/polynote/polynote/assets/5430417/4da3fd90-778a-4c8c-afae-7167a5e8a013)

Reply:
.....oops, yeah, that does exactly what I want. Will refactor, thank you!